From fffe7983af9786e48f8b7d762a1194cbf0acc743 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tom=C3=A1=C5=A1=20Kelemen=20=28vudiq=29?= <vudiq@vudiq.sk>
Date: Fri, 24 May 2024 19:50:37 +0200
Subject: [PATCH] remove fflush() call before exit()

the buffer is flushed anyways at exit() and fixes compile error with
gcc14:

    pico/carthw/svp/compiler.c: In function 'ssp_translate_block':
    pico/carthw/svp/compiler.c:1800:24: error: passing argument 1 of 'rfflush' from incompatible pointer type [-Wincompatible-pointer-types]
     1800 |                 fflush(stdout);
          |                        ^~~~~~
          |                        |
          |                        FILE *
    In file included from ./pico/pico_port.h:12,
                     from ./pico/pico_int.h:15,
                     from pico/carthw/svp/compiler.c:9:
    platform/libretro/libretro-common/include/streams/file_stream_transforms.h:89:25: note: expected 'RFILE *' but argument is of type 'FILE *'
       89 | int64_t rfflush(RFILE * stream);
          |                 ~~~~~~~~^~~~~~

this error presents itself when building the libretro core, so this
could be also solved by wrapping the call to fflush():

    #ifndef __LIBRETRO__
    		fflush(stdout);
    #else
                fflush((RFILE *)stdout);
    #endif
---
 pico/carthw/svp/compiler.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/pico/carthw/svp/compiler.c b/pico/carthw/svp/compiler.c
index c472a6008..941187ce6 100644
--- a/pico/carthw/svp/compiler.c
+++ b/pico/carthw/svp/compiler.c
@@ -1797,7 +1797,6 @@ void *ssp_translate_block(int pc)
 
 	if (tcache_ptr - (u32 *)tcache > DRC_TCACHE_SIZE/4) {
 		elprintf(EL_ANOMALY|EL_STATUS|EL_SVP, "tcache overflow!\n");
-		fflush(stdout);
 		exit(1);
 	}
 
